simpleocr/files.py

```diff
@@ -45,45 +32,152 @@ def write(self):
         write_boxfile(self.path, self.classes, self.segments)
 
 
-class ImageFile(object):
```

Comment:
To make a direct comparison possible, I wrote the actual classes (though only in draft state at the moment, it didn't take much time at all) and then accidently committed them to this branch 😒 . I reset the changes and then pushed them to another branch, that can now be found [here](https://github.com/RedFantom/simple-ocr-opencv/tree/image_classes/simpleocr) (all the class properties are to emphasize the immutability to the user by the way). Actually, it doesn't really make handling the objects in `ocr.py` any harder at all, because `isinstance` supports child-classes as well. Therefore, this actually seems like a good solution, and if we document the two classes well enough (I can update the wiki to reflect the changes) it would actually be just as clear as with the single class. Also, the code is backwards-compatible with the earlier code (except for the now separate path finding part). It would also allow for much easier work in the `improver.py` file if we get to it, actually, as it is then possible to not modify the original `ImageFile` object of the user but simply return a new `ImageBuffer` object.